apps/kafka-producers/templates/deployment.yaml

```diff
@@ -86,6 +86,15 @@ spec:
               {{- else }}
               value: {{ $.Values.env.logLevel | quote }}
               {{- end }}
+            {{- if $.Values.existingSecret }}
+            - name: TS_SALKAFKA_USERNAME
```

Comment:
Please don't hardcode the username here. There is a configuration parameter (`env.extras`) that allows one to specify extra environment variables.